workers/unity/Packages/com.improbable.gdk.core/Worker/SpatialOSWorld.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Collections.Generic;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Improbable.Gdk.Core
+{
+    public class SpatialOSWorld : World
+    {
+        private WorkerBase worker;
+        private List<ScriptBehaviourManager> spatialOSSystemManagers;
+        public EntityGameObjectLinker EntityGameObjectLinker { get; }
+
+        public SpatialOSWorld(string name) : base(name)
+        {
+            EntityGameObjectLinker = new EntityGameObjectLinker(this);
+        }
+
+        public void Connect(ConnectionConfig config, Vector3 origin)
+        {
+            // Create worker, connect to fabric and register worker-specific systems
+            worker = WorkerRegistry.CreateWorker(config, origin);
+            foreach (Type type in worker.Systems)
+            {
+                var manager = GetOrCreateManager(type);
+                spatialOSSystemManagers.Add(manager);
+            }
+            WorkerRegistry.SetWorkerForWorld(worker, this);
+        }
+
+        public void Disconnect(string reason)
+        {
+            foreach (ScriptBehaviourManager manager in spatialOSSystemManagers)
+            {
+                DestroyManager(manager);
+                WorkerRegistry.UnsetWorkerForWorld(worker, this);
+            }
+            Dispose();
+        }
+
+        public new void Dispose()
```

Comment:
What will happen if you call Dispose but don't call Disconnect while still connected? It doesn't make sense for the user to do the following:

```
var world = new SpatialOSWorld("Foo");
world.Connect();
world.Disconnect();
world.Dispose();
```

If you try this then you will get an exception on line 44 because Disconnect will call Dispose which sets the worker to null and then manually calling `world.Dispose` will try to do `worker.Dispose()`.

Given that the worker is initialized in Connect it makes sense to destroy it in Disconnect and only clear things in Dispose which were initialized in the constructor.

Reply:
as said: this is currently not working at all ;)
but good point and I'll take it into account for the next PR